### PR TITLE
Add RecognizeEntities to Azure.AI.TextAnalytics

### PR DESCRIPTION
**Pull request by nisha-bhatia**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs
Comment (jsquire) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:768]:
```suggestion
                var input = new MultiLanguageInput(id.ToString(), document);
```

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs
Comment (jsquire) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:668]:
```suggestion
            return await RecognizeEntitiesBatchAsync(input, options, cancellationToken).ConfigureAwait(false);
```

**Review by jsquire — COMMENTED** [review]
Looks reasonable to me.   I'll defer to @maririos for approval.

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:763]: Why do you pass a newly created object `AnalyzeTextEntityRecognitionInput textEntityInputs` to then populate it and return it?
Why not just create it here in this function and then return it?

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:703]: `entityRecognitionInput` was already populated in the `TextDocumentInputToEntityRecognition`. There is no need to create a new `AnalyzeTextEntityRecognitionInput` and assign all the values again

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs:199]: you already have a method to do this https://github.com/Azure/azure-sdk-for-net/blob/871dc09e0da54714390e0e4f6b0ce2d76ca1940c/sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs#L83

No need to do it again.

Comment (jsquire) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs:199]: dang....  I missed this.   I'm a dirty hack who will go sit in the corner now.

Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs:199]: No need :( fresh eyes always help

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs
Comment (nisha-bhatia) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:763]: I think I did the same thing in LanguageDetection, I'll update that method as well

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:697]: this has not been resolved, and this is actually adding a bug because `StringIndexType` is never set.
Let me know if you have questions about my comment

Comment (nisha-bhatia) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:697]: I think I set it in line 768. StringIndexType is the last parameter

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs
Comment (nisha-bhatia) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:768]: StringIndexType is set here

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:697]: Correct, but look at the flow of this function:
- You get as a parameter `AnalyzeTextEntityRecognitionInput entityRecognitionInput` which indeed was correctly set.
- In line 705  you create a new `AnalyzeTextEntityRecognitionInput` called `analyzeRecognizeEntities ` and assign to it `entityRecognitionInput.AnalysisInput`
- line 706 and 707 you assign to `analyzeRecognizeEntities` parameters that don't include `StringIndexType`
- line 710 you are passing to the `AnalyzeAsync` method the `analyzeRecognizeEntities` that you created in this function

So in the end, you are not passing the pretty `entityRecognitionInput` that you created before.

My point is, you already did all the work to correctly populate `entityRecognitionInput`, why not use that one directly for the `AnalyzeAsync` method?

Comment (nisha-bhatia) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:697]: sorry, I understood it now. updated

Comment (nisha-bhatia) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:697]: I was using a lot of extra memory before

**Review by maririos — APPROVED** [review]
LGTM.
The conflict was probably added after I merged my tests PR.

**Pull request merged**
